container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1132,6 +1134,10 @@ func (manager *KubernetesManager) RemovePod(ctx context.Context, pod *apiv1.Pod)
 		return stacktrace.Propagate(err, "Failed to delete pod with name '%s' with delete options '%+v'", name, globalDeleteOptions)
 	}
 
+	if err := manager.waitForPodTermination(ctx, namespace, name); err != nil {
```

Comment:
We need to be more careful around pod termination now that we are going to support stop service.  This is even beneficial today.